src/Nethermind/Nethermind.Consensus/Producers/TxPoolTxSource.cs

```diff
@@ -62,7 +62,12 @@ public IEnumerable<Transaction> GetTransactions(BlockHeader parent, long gasLimi
 
             int checkedTransactions = 0;
             int selectedTransactions = 0;
-            using ArrayPoolList<Transaction> selectedBlobTxs = new(_eip4844Config.GetMaxBlobsPerBlock());
+            if (spec.IsEip7742Enabled && payloadAttributes == null)
+            {
+                if (_logger.IsDebug) _logger.Debug("Eip-7742 is enabled, but no payload attributes were provided.");
+                yield break;
+            }
```

Comment:
is this true? if yes, should throw exception or yield break is okay?

Reply:
Good question, we could just build return transactions ingoring blobs